packages/block-library/src/social-links/edit.js

```diff
@@ -131,51 +130,48 @@ export function SocialLinksEdit( props ) {
 						isAlternate: true,
 					} }
 				/>
-				<ToolbarGroup>
-					<ToolbarItem>
-						{ ( toggleProps ) => (
-							<DropdownMenu
-								label={ __( 'Size' ) }
-								text={ __( 'Size' ) }
-								icon={ null }
-								popoverProps={ POPOVER_PROPS }
-								toggleProps={ toggleProps }
-							>
-								{ ( { onClose } ) => (
-									<MenuGroup>
-										{ sizeOptions.map( ( entry ) => {
-											return (
-												<MenuItem
-													icon={
-														( size ===
-															entry.value ||
-															( ! size &&
-																entry.value ===
-																	'has-normal-icon-size' ) ) &&
-														check
-													}
-													isSelected={
-														size === entry.value
-													}
-													key={ entry.value }
-													onClick={ () => {
-														setAttributes( {
-															size: entry.value,
-														} );
-													} }
-													onClose={ onClose }
-													role="menuitemradio"
-												>
-													{ entry.name }
-												</MenuItem>
-											);
-										} ) }
-									</MenuGroup>
-								) }
-							</DropdownMenu>
-						) }
-					</ToolbarItem>
-				</ToolbarGroup>
+				<ToolbarItem>
```

Comment:
As far as I understand, `BlockControlSlot` already returns a `ToolbarGroup` and should not be necessary anymore.

Not sure if the Size dropdown should live in the `Other` section instead of the block one. It seems to me it belongs in `Block` as it's a formatting option, but other options that don't have an icon but a text are usually in `Other` (as in the Image block)